facebook/micro/monthly-archive.sh

```diff
@@ -8,13 +8,20 @@ else
 fi
 echo ${MONTH}
 R_MONTH=${MONTH#*_}; R_MONTH=${R_MONTH#0}
-BATCH="cd fb-public-results\nls -1 cvid_responses_${MONTH}*.gz"
-sftp -b <(echo -e "${BATCH}") -P 2222 fb-automation@ftp.delphi.cmu.edu 2>/dev/null | \
-    grep "^cvid" | \
-    awk -F_ 'BEGIN{print "cd fb-public-results"} {key=$3 $4 $5; if (key!=last && last!="") {print record} last=key; record=$0} END{print record}' | \
-    sed '/^cvid/ s/^/get /' >fetch.sftp
-sftp -b fetch.sftp -P 2222 fb-automation@ftp.delphi.cmu.edu
-OUT=${MONTH/_/-}
-Rscript ../monthly-files.R ${MONTH%_*} ${R_MONTH} . >${OUT}.csv
-gzip ${OUT}.csv
-sftp -b <(echo -e "cd fb-public-results\nput ${OUT}.csv.gz") -P 2222 fb-automation@ftp.delphi.cmu.edu
+
+perform_rollup_and_post ()
+{
+    BATCH="cd $1\nls -1 cvid_responses_${MONTH}*.gz"
+    sftp -b <(echo -e "${BATCH}") -P 2222 fb-automation@ftp.delphi.cmu.edu 2>/dev/null | \
+        grep "^cvid" | \
+        awk -F_ 'BEGIN{print "cd '"$1"'"} {key=$3 $4 $5; if (key!=last && last!="") {print record} last=key; record=$0} END{print record}' | \
+        sed '/^cvid/ s/^/get /' >fetch.sftp
+    sftp -b fetch.sftp -P 2222 fb-automation@ftp.delphi.cmu.edu
+    OUT=${MONTH/_/-}
+    Rscript ../monthly-files.R ${MONTH%_*} ${R_MONTH} . >${OUT}.csv
+    gzip ${OUT}.csv
+    sftp -b <(echo -e "cd $1\nput ${OUT}.csv.gz") -P 2222 fb-automation@ftp.delphi.cmu.edu
+}
```

Comment:
This script was originally designed to leave all the .gz files in place, making it easy to debug if something went wrong. It cleans up all .gz files at the beginning of the script. Since we're running it twice, and presumably want `monthly-files.R` to see a different set of files for each run, _but_ we still want to retain the .gz files for debugging, we need to clean up after ourselves instead:
```suggestion
    rm -rf $1
    mkdir $1
    mv *.gz $1/
}
```